src/treemap.ts

```diff
@@ -55,11 +50,19 @@ export function isDOMNode(e: Element): boolean {
  */
 export interface Options {
   padding: [number, number, number, number];
-  caption?(node: Node): string;
+  lowerBound: number;
+  classList: ((node: Node) => string)[];
+  nodeColor: (node: Node) => NodeColor;
+  caption(node: Node): string;
   showNode(node: Node, width: number, height: number): boolean;
   showChildren(node: Node, width: number, height: number): boolean;
 }
 
+interface NodeColor {
+  defaultColor: string;
+  hoverColor: string;
+}
```

Comment:
I wonder, can we achieve this with CSS?

Reply:
I wondered the same thing. My initial attempt to do it this way was how the `classList` option was born. 

Because `:hover` is a pseudo selector, it can't be changed with any DOM APIs. To achieve this with css you would have to inject style rules programatically into the page and have them specifically target certain nodes based on the callback you pass in.

Imagine a set of data where we want the node with the greatest size to be red and every other node a lighter shade of red proportional to the ratio of its size to the largest node's size.

I could not think of a css solution to this case so I decided to go with the `mouseleave` and `mouseenter` events to edit the style of each DOM element inline. I would love to discuss alternate solutions if you have any in mind.

Reply:
Wow, it is disappointing that you can't do this with inline style!
Is there some way we can leave it up to the caller to do CSS tweaks other than the hover?  E.g. via the classlist thing?

Reply:
I've had a moment to reflect on this and agree that the current solution is not ideal. The use case I describe could potentially be solved on the user end using more advanced css (scss @for loops for example) and the classList option. I appreciate you calling this out, sometimes these things have to be discussed for an ideal solution to be reached.

I'll remove this option in favour of the user defined class approach. I suspect this would greatly improve performance as well for larger trees.